packages/zoe/src/zoeService/refillMeter.js

```diff
@@ -0,0 +1,23 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { makeAsyncIterableFromNotifier as iterateNotifier } from '@agoric/notifier';
+
+/**
+ * @param {Meter} meter
+ * @param {ChargeForComputrons} chargeForComputrons
+ * @param {ERef<FeePurse>} feePurse
+ */
+const refillMeter = async (meter, chargeForComputrons, feePurse) => {
+  const meterNotifier = E(meter).getNotifier();
+
+  for await (const remaining of iterateNotifier(meterNotifier)) {
+    const incrementBy = await chargeForComputrons(feePurse);
```

Comment:
Attn @michaelfig @warner @Chris-Hibbert @kriskowal 

This `await` is interesting. It violates our syntactic `await` safety rule, but, AFAICT, it does not violate the purpose of that rule. I think that is because the top level of a `for await of` *specifically* (and not any other looping construct) treats its body enough like a separately called closure.

No change suggested at this time. But we should either
   * satisfy ourselves that this one is safe, and revise our advise and our lint checks accordingly, or
   * fail to satisfy ourselves that it is safe, and revise the code here to be safe.

Note that the obvious rewrite would be to use `observeNotifier` instead. But this would not be as good, because it wouldn't wait for the body-equivalent (the `updateState` method of the observer) before delivering another notification, and so would notify more often than needed.